src/libcore/result.rs

```diff
@@ -684,11 +684,16 @@ impl<T, E: fmt::Debug> Result<T, E> {
     pub fn unwrap(self) -> T {
         match self {
             Ok(t) => t,
-            Err(e) =>
-                panic!("called `Result::unwrap()` on an `Err` value: {:?}", e)
+            Err(e) => Self::unwrap_failed(e),
         }
     }
 
+    #[inline(never)]
+    #[cold]
+    fn unwrap_failed(error: E) -> ! {
+        panic!("called `Result::unwrap()` on an `Err` value: {:?}", error)
+    }
```

Comment:
One other possible option you could play around with is seeing if this improves codegen by moving it outside the `impl` block and taking `&fmt::Display` to ensure it's just a small function call